code/__HELPERS/icons.dm

```diff
@@ -692,7 +692,18 @@ The _flatIcons list is a cache for generated icon files.
 	var/image/copy
 	// Add the atom's icon itself, without pixel_x/y offsets.
 	if(!noIcon)
-		copy = image(icon=curicon, icon_state=curstate, layer=A.layer, dir=curdir)
+		if(curdir == SOUTH)
+			copy = image(curicon, curstate)
+		else
+			var/icon/dirIcon
+			if(curdir == NORTH || curdir == SOUTH || curdir == EAST || curdir == WEST)

```

Comment:
This is usually done like ``if(curdir in cardinals)``, it's very slightly faster too.